internal/github/auth.go

```diff
@@ -0,0 +1,46 @@
+package github
+
+import (
+	"fmt"
+	"github.com/pkg/errors"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+const (
+	envAuth       = "GITHUB_TOKEN"
+	authTokenFile = ".elastic/github.token"
+)
+
+// EnsureAuthConfigured method ensures that GitHub auth token is available.
+func EnsureAuthConfigured() error {
+	_, err := AuthToken()
+	if err != nil {
+		return errors.Wrapf(err, "GitHub authorization token is missing. Please use either environment variable %s or ~/%s",
+			envAuth, authTokenFile)
+	}
+	return nil
+}
+
+// AuthToken method finds and returns the GitHub authorization token.
+func AuthToken() (string, error) {
+	githubTokenVar := os.Getenv(envAuth)
+	if githubTokenVar != "" {
+		fmt.Println("Using GitHub token from environment variable.")
```

Comment:
Not for this PR but at some point later we should consider introducing proper logging into this tool.

Reply:
ECS logging lib ;-)

Reply:
https://github.com/elastic/elastic-package/issues/45